### PR TITLE
UCP: fallback into single thread on single-thread build

### PR DESCRIPTION
**Pull request by hoopoepg**
- in case if single thread build is used - force single thread
  worker initialization instead of error

### CODE REVIEW

File: src/ucp/core/ucp_worker.c [hunk 1]
```diff
@@ -1366,11 +1366,10 @@ ucs_status_t ucp_worker_create(ucp_context_h context,
 
     if (params->field_mask & UCP_WORKER_PARAM_FIELD_THREAD_MODE) {
 #if !ENABLE_MT
-        if (params->thread_mode == UCS_THREAD_MODE_MULTI) {
-            return UCS_ERR_INVALID_PARAM;
-        }
-#endif
+        thread_mode = UCS_THREAD_MODE_SINGLE;
```
Comment (shamisp) [inline src/ucp/core/ucp_worker.c]: shell we issue some warring is we simply ignore/overwrite thread_mode ?

Comment (hoopoepg) [reply src/ucp/core/ucp_worker.c]: don't think so - the behavior is same as for MPI_Init_thread - requested thread level may be different from supported thread level

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8502/ for details (Mellanox internal link).

File: src/ucp/core/ucp_worker.c (hunk 1 above)
Comment (yosefe) [reply src/ucp/core/ucp_worker.c]: @hoopoepg let's add API documentation to ucp.h worker_create and ucp_init that the actual thread mode may differe than requested, and user should use query to find the actual thread mode.

Comment (yosefe) [reply src/ucp/core/ucp_worker.c]: we can also add debug print (but not warning/error)

Comment (hoopoepg) [reply src/ucp/core/ucp_worker.c]: @yosefe docs are updated

Comment (yosefe) [reply src/ucp/core/ucp_worker.c]: thanks! do we need to update the doc of ucp_init() as well?

Comment (hoopoepg) [reply src/ucp/core/ucp_worker.c]: why not... will add another commit

Comment (hoopoepg) [reply src/ucp/core/ucp_worker.c]: @yosefe updated desc for ucp_param_t

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8503/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8505/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8512/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
bot:mlx:retest

**Comment by bbenton** [comment]
:+1: from Doc perspective

**Comment by shamisp** [comment]
👍

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8513/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8522/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8534/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
@yosefe ok to merge?

**Pull request merged**